controller/lib/core/blower_pid.cpp

```diff
@@ -43,76 +38,28 @@ static PID myPID(Kp, Ki, Kd, ProportionalTerm::ON_ERROR,
                  // Our output is an 8-bit PWM.
                  /*output_min=*/0.f, /*output_max=*/255.f, PID_SAMPLE_PERIOD);
 
-// TODO: VOLUME_INTEGRAL_INTERVAL was not chosen carefully.
-static constexpr Duration VOLUME_INTEGRAL_INTERVAL = milliseconds(5);
-static Time last_flow_measurement_time = millisSinceStartup(0);
-static float last_flow_ml_per_min = NAN;
-
 void blower_pid_init() {
 }
 
-static void update_volume(SensorReadings *readings) {
-  // TODO: This calculation should be much more sophisticated.  Some possible
-  // improvements.
-  //
-  //  - Periodically re-zero the volume (e.g. what happens if the tubes are
-  //    disconnected from the patient?)
-  //
-  //  - Measure time with better than millisecond granularity.
-  Time now = Hal.now();
-  if (isnan(last_flow_ml_per_min)) { // First time
-    readings->volume_ml = 0;
-    last_flow_ml_per_min = readings->flow_ml_per_min;
-    last_flow_measurement_time = now;
-  } else if (Duration delta = now - last_flow_measurement_time;
-             delta >= VOLUME_INTEGRAL_INTERVAL) {
-    readings->volume_ml += delta.minutes() *
-                           (last_flow_ml_per_min + readings->flow_ml_per_min) /
-                           2;
-    last_flow_ml_per_min = readings->flow_ml_per_min;
-    last_flow_measurement_time = now;
-  }
-}
-
-void blower_pid_execute(const BlowerSystemState &desired_state,
-                        SensorReadings *readings, float *fan_power) {
-  // Open/close the solenoid as appropriate.
-  //
-  // Our solenoid is "normally open", so low voltage means open and high
-  // voltage means closed.  Hardware spec: https://bit.ly/3aERr69
-  Hal.digitalWrite(BinaryPin::SOLENOID,
-                   desired_state.expire_valve_state == ValveState::OPEN
-                       ? VoltageLevel::HAL_LOW
-                       : VoltageLevel::HAL_HIGH);
-
-  Pressure cur_pressure = get_patient_pressure();
-
+float blower_pid_compute_fan_power(Time now,
+                                   const BlowerSystemState &desired_state,
+                                   const SensorReadings &sensor_readings) {
   // If the blower is not enabled, immediately shut down the fan.  But for
   // consistency, we still run the PID iteration above.
   float output;
   if (desired_state.blower_enabled) {
-    output = myPID.Compute(Hal.now(), /*input=*/cur_pressure.kPa(),
-                           /*setpoint=*/desired_state.setpoint_pressure.kPa());
+    output =
+        myPID.Compute(/*time=*/now,
```

Comment:
Still remains - whether in this PR or not - to change this from (function using a static myPID) to (static instance of a class that has myPID as a member variable).